planner/core/logical_plan_builder.go

```diff
@@ -557,18 +557,29 @@ func (b *PlanBuilder) buildProjectionFieldNameFromColumns(field *ast.SelectField
 }
 
 // buildProjectionFieldNameFromExpressions builds the field name when field expression is a normal expression.
-func (b *PlanBuilder) buildProjectionFieldNameFromExpressions(field *ast.SelectField) model.CIStr {
+func (b *PlanBuilder) buildProjectionFieldNameFromExpressions(field *ast.SelectField) (model.CIStr, error) {
 	if agg, ok := field.Expr.(*ast.AggregateFuncExpr); ok && agg.F == ast.AggFuncFirstRow {
 		// When the query is select t.a from t group by a; The Column Name should be a but not t.a;
-		return agg.Args[0].(*ast.ColumnNameExpr).Name.Name
+		return agg.Args[0].(*ast.ColumnNameExpr).Name.Name, nil
 	}
 
 	innerExpr := getInnerFromParenthesesAndUnaryPlus(field.Expr)
+	funcCall, isFuncCall := innerExpr.(*ast.FuncCallExpr)
+	// When used to produce a result set column, NAME_CONST() causes the column to have the given name.
+	// See https://dev.mysql.com/doc/refman/5.7/en/miscellaneous-functions.html#function_name-const for details
+	if isFuncCall && funcCall.FnName.L == ast.NameConst {
+		if v, err := evalAstExpr(b.ctx, funcCall.Args[0]); err == nil {
```

Comment:
Return error also if `err` is not nil?

Reply:
yes, I make a mistake here. line 575 should be moved outside the curly brace in line 576.

Reply:
seems we should directly return error as long as the first parameter is not a constant:
```sql
MySQL(root@localhost:test) > select name_const(1+1, 1) from t;
ERROR 1210 (HY000): Incorrect arguments to NAME_CONST
```

Reply:
It is guaranteed aster preprocess: https://github.com/pingcap/tidb/pull/9261/files#diff-c95e0584bc651e7f623fb47d69f690bcR143. So we don't need to call `evalAstExpr()` to get the constant value.

Reply:
@spongedu PTAL this comment. Could you abandon the function call of `evalAstExpr()`?

Reply:
@zz-jason for `name_const`, we should get the first argument to rename the output column, so I think `evalAstExpr` is still needed?